src/vault.rs

```diff
@@ -135,6 +144,20 @@ impl Client {
         Ok(())
     }
 
+    pub fn kubernetes_auth(&mut self, role: String) -> Result<(), Error> {
+        let mut file = File::open("/var/run/secrets/kubernetes.io/serviceaccount/token").expect("Unable to open");
+        let mut jwt = String::new();
+        file.read_to_string(&mut jwt);
```

Comment:
Both `File::open` and `File::read_to_string` can return a `std::io::Error`, to be able to use the `?` operator you need to be able to convert a `std::io::Error` to the `Error` type this function returns (defined up on line 87).

To be able to do that conversion you need to implement the `std::convert::From` trait. Currently the way `Error` is defined it holds a `ClientError`, but you've got a `std::io::Error`, so you probably want it to also be able to hold a `ClientError` *or* a `std::io::Error`. You would do this by changing it to an enum, like:

``` rust
pub enum Error {
    ClientError(ClientError),
    IoError(std::io::Error),
}
```

You'd then need to update the various trait implementations for `ClientError` to now be able to cope with it being an enum, and finally implement the `From` trait. Check [`client_error.rs`] for examples.

[`client_error.rs`]: https://github.com/avvo/avvoenv/blob/master/src/client_error.rs

---

Ignoring the error from `File::read_to_string` will mean that if it fails you'll end up with an empty `jwt`, and things will probably break later down the line in confusing ways.

Using `expect(...)` on the error from `File::open` will mean the program will immediately crash with an ugly error, instead of passing the error back up to any error handing that may exist in the caller.